lib/src/tar/tar_file.dart

```diff
@@ -223,16 +221,16 @@ class TarFile {
   }
 
   void _writeString(OutputStream output, String value, int numBytes) {
-    final codes = List<int>.filled(numBytes, 0);
-    final end = numBytes < value.length ? numBytes : value.length;
+    final codes = Uint8List(numBytes);
+    final end = min(numBytes, value.length);
     codes.setRange(0, end, value.codeUnits);
     output.writeBytes(codes);
   }
 
   void _writeInt(OutputStream output, int value, int numBytes) {
     var s = value.toRadixString(8);
     while (s.length < numBytes - 1) {
-      s = '0' + s;
+      s += '0';
```

Comment:
My 'pad with 0s' loop here isn't great. But this change will be appending instead of prepending the 0s.

Reply:
Probably the reason why the build failed